src/lib/throttle.ts

```diff
@@ -0,0 +1,19 @@
+import { throttle as _throttle } from 'lodash';
+
+/**
+ * Creates a curried function for throttling a function
+ *
+ * default returns a instance of lodash's throttle function with a function and wait time set.
+ * This throttle handles promise rejections better then a regular lodash throttle by allowing
+ * the caller to handle the exception instead of lodash.
+ */
+export default function throttle(
+	fn: (opts: any) => Promise<void>,
```

Comment:
I'd like to think about how to make the fn arguments more generic

Reply:
You could use generics and all type checks should work
```
export default function throttle<T>(
	fn: (opts: T) => Promise<void>,
```

Reply:
Having this function in lib/throttle seems to imply that it should be used over directly using lodash's throttle. What happens if you pass a function to throttle that's not a Promise?

Reply:
If you want to nicely inherit the args of the function regardless of what they are you'd be better off with something like
```ts
export default function throttle<T extends (...args: any[]) => Promise<void>>(
	fn: T,
	wait: number,
): (
	args: Parameters<T>,
	resolve: () => void,
	reject: (e: Error) => void,
) => void {
	return _throttle(
		(
			args: Parameters<T>,
			resolve: () => void,
			reject: (e: Error) => void,
		) => {
			fn(...args)
				.then(resolve)
				.catch(reject);
		},
		wait,
	);
}
```